From 3a9906d2b5d1c504670442277906ed53cd9686e8 Mon Sep 17 00:00:00 2001
From: Dario Freddi <drf@kde.org>
Date: Fri, 25 Nov 2011 23:08:07 +0100
Subject: [PATCH 20/20] CCMAIL: echidnaman@kubuntu.org

KCrash must be initialized after KUniqueApplication is started, to avoid KCmdLine to parse its arguments before that.

P.S.: While I do appreciate the contribution and I think the patch was really useful,
it would be awesome if those changes went through review first. If they went through and I forgot
about it and didn't spot the problem earlier, feel free to hit me with a stick instead.
---
 main.cpp |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/main.cpp b/main.cpp
index ab359b7..33ef236 100644
--- a/main.cpp
+++ b/main.cpp
@@ -35,13 +35,13 @@ int main(int argc, char *argv[])
 
     KCmdLineArgs::init(argc, argv, &aboutData);
 
-    KCrash::setFlags(KCrash::AutoRestart);
-
     if (!PolicyKitKDE::start()) {
         qWarning("PolicyKitKDE is already running!\n");
         return 0;
     }
 
+    KCrash::setFlags(KCrash::AutoRestart);
+
     PolicyKitKDE agent;
     agent.disableSessionManagement();
     agent.exec();
-- 
1.7.7.4

